frappe/integrations/doctype/google_calendar/google_calendar.py

```diff
@@ -517,10 +517,10 @@ def google_calendar_to_repeat_on(start, end, recurrence=None):
 	repeat_on = {
 		"starts_on": get_datetime(start.get("date"))
 		if start.get("date")
-		else parser.parse(start.get("dateTime")).utcnow(),
+		else parser.parse(start.get("dateTime")).astimezone().replace(tzinfo=None),
```

Comment:
@ankush @this-gavagai

Not sure if I missed something, but it seems like the current code converts to the timezone of the server, not to the timezone according to **System Settings**?

Maybe the following would work (untested):

```python
from zoneinfo import ZoneInfo

parser.parse(start.get("dateTime")).astimezone(ZoneInfo(get_time_zone())).replace(tzinfo=None)
```

Reply:
I had it like that originally, Raffael, but it was giving me incorrect results. When I used server time, it seemed to be working, and at the time I interpreted that as a function of how the date field works. I believe you are correct, though. I'll run a few tests and update the patch.

Reply:
I've followed up with a new PR (#18458) per your suggestions, @barredterra.

I'm still very unclear about the interaction between System Settings, User Settings, and server configs, but this PR seems robust empirically. There may definitely be configurations that I haven't thought to test, however.

I am also working on some bugs in how repeating events are handled. Once that is done, I'll set up some unit tests as well.